src/extension.ts

```diff
@@ -31,6 +31,65 @@ export function activate(context: vscode.ExtensionContext) {
 	});
 	context.subscriptions.push(disposable_number_cursors_from_one);
 	
+	let disposable_number_cursors_from_arbitrary = vscode
+			.commands
+			.registerTextEditorCommand('engineernick.multi-cursor-tools.number_cursors_from_arbitrary', (editor, edit, args) => {
+		vscode.window.showInputBox({
+			placeHolder: "1",
+			prompt: "Starting number",
+			value: 1
+		}).then((num)=>{
+			if(parseFloat(num).toString() == num){
```

Comment:
yarn lint complains here because num may be undefined. Perhaps if the user wants to cancel the operation. Instead of logging an error I recommend some feedback to the user like;
```suggestion
                        if (user_input === undefined || user_input === "") {
                            vscode.window.showErrorMessage('A number was required. No changes made.');
                            return
                        }
                        let num = parseFloat(user_input);
                        if(Number.isSafeInteger(num) && num > 0 && num < 10000){
```

Probably this could be a bit more elegant with error messages / if blocks but the suggestions feature of github reviews is actually pretty fiddly :P

Reply:
This one ... I fully agree with the first test and message ... however ... I would still prefer having 
```js
num.toString() === user_input
```
over
```js
Number.isSafeInteger(num) && num > 0 && num < 10000
```
I don't actually see the need for limiting to strictly _(0 with steps seems a pretty reasonable need)_ positive _(starting at -10 seems also possible)_ numbers neither to integers _(why not give possibility to use floats? ... since it makes it not more complicated in my opinion)_ 

... I could understand the need to limit the maximal (absolute) value _[however don't see the need to be that limiting : 10'000]_ ... do you fear overflow problems ?
=> I added a check against `Number.MAX_SAFE_INTEGER` ... seemed reasonable to me

The advantage of that solution being the flexibility : if I enter an int => gonna work just fine ! Same for a float ! But if anything else ... it'll refuse user_input (agreed however : we should display an error to the user => implemented that following your guide above)

I've actually read quite a bit about checking numbers before this PR since I wanted to do things as good as possible ... and found the above being the most elegant to me while keeping maximum functionnality... 

I however do understand it is your codebase ... and would fully understand if you were to prefer your tests ... this is the reason why I corrected and implemented messages with your tests on a second branch : 
[ADD-possibility-to-number-mutli-cursors-choosing-steps-Nick-tests](https://github.com/kjo-sdds/nick-says-vs-code-extension/tree/ADD-possibility-to-number-mutli-cursors-choosing-steps-Nick-tests)

This _("main")_ branch contains the changes as I see them best ... but feel free to dismiss this PR and make a new one from the aforementionned branch.

About style ... I would rather have a guarding clause before (makes sense in particular with the steps choosing option ... since for now the error message is ... pretty far ... due to nesting ... ) but the price for that is a negative test (to stay as easily readableas possible) which to me was "affordable enough" that I chose this path ...